ovsdb/client.go

```diff
@@ -146,9 +146,16 @@ func (c *Client) rpc(ctx context.Context, method string, out interface{}, args .
 	// Await RPC completion or cancelation.
 	select {
 	case <-ctx.Done():
-		// RPC canceled; clean up callback.
-		return c.cancelCallback(ctx, req.ID)
-	case res := <-ch:
+		// RPC canceled.  Producer cleans up the callback.
+		return ctx.Err()
+	case res, ok := <-ch:
+		if !ok {
+			// Channel was closed by producer after a context cancelation,
+			// and woke up this consumer.  The select statement happened
+			// to pick this case even though the context was canceled.
+			return ctx.Err()
```

Comment:
Like it!  Thanks @mdlayher, and love the multiple commit to let us fucus the specific change.

LGTM!